File: Q1-20/Q1/Q1.js
var twoSum = function(nums, target) {
    // 84ms; 39.9MB
    const num_dict = {}
    let result = []
    nums.forEach((k, i) => {
        if (target - k in num_dict) {
            result.push(num_dict[target - k], i)
        } else {
            num_dict[[k]] = i
        }
    })
    return result
};




console.log(twoSum(nums = [2,7,11,15], target = 9))

File: Q1-20/Q1/Q1.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var numDict = [Int: Int] ()
        for (i, k) in nums.enumerated() {
            if let i2 = numDict[target - k] {
                return [i2, i]
            } else {
                numDict[k] = i
            }
        }
        return [0, 1]
    }
}

print(Solution().twoSum([2,7,11,15], 9))
